### ISSUE TITLE
Connection used in RoutingFilter can not be released in specific case

### ISSUE THREAD

**Issue by iqjumps** [spring-cloud/spring-cloud-netflix#1790]
What I understand is:
RoutingFilter -> sendErrorFilter -> SendResponseFilter
After RoutingFilter routes request, ResponseDataStream is used in SendResponseFilter
Connection can be released when writeResponse() executes (when inputstream meet eof, connection released)

But I got exception when writeResponse() in SendResponseFilter
``` OutputStream outStream = servletResponse.getOutputStream(); ```
It is because of getOutputStream() in org.apache.catalina.connector.Response
``` 
if (usingWriter) {
    throw new IllegalStateException
	(sm.getString("coyoteResponse.getOutputStream.ise"));
}
```
When error case, SendErrorFilter forwards request and some handler uses it's outputstream and 'usingWriter' is set to true
So when SendReponseFilter tries to open outputStream, it throws exception

in my case, I requested text/html and render() in ErrorMvcAutoConfiguration opens writer
```
@Override
public void render(Map<String, ?> model, HttpServletRequest request,
		HttpServletResponse response) throws Exception {
	if (response.getContentType() == null) {
		response.setContentType(getContentType());
	}
	Map<String, Object> map = new HashMap<String, Object>(model);
	map.put("path", request.getContextPath());
	PlaceholderResolver resolver = new ExpressionResolver(this.expressions, map);
	String result = this.helper.replacePlaceholders(this.template, resolver);
	response.getWriter().append(result);
}
```
After request just 20 times(it's default max connection per route), all my request are freezing (They wait to get connection from pool but can not get forever)

How can I solve this problem?

Sorry that I'm not fluent to englishTT

**Comment by spencergibb** [comment]
What version are you using?

**Comment by iqjumps** [comment]
I'm using 1.1.5.RELEASE

**Comment by spencergibb** [comment]
I believe this may have been fixed here 3e7b196950c5f1aec6899fc35089b7c7be7bc72d

**Comment by spencergibb** [comment]
Closing this due to inactivity. Please re-open if there's more to discuss.
